### ISSUE TITLE
testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews flakey

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#55778]
In scenario app
```
TextSemanticsFocusTest
2020-04-27 14:15:11.169 xcodebuild[66853:298993] [MT] DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTiOSFrameworks/DVTiOSFrameworks-14802/IDEiOSSupportCore/DVTiPhoneSimulator.m:1890
Details:  [DVTiPhoneSimulator -primaryInstrumentsServer was called from the main thread
Object:   <DVTiPhoneSimulator: 0x7fddc5226240>
Method:   -primaryInstrumentsServerWithError:
Thread:   <NSThread: 0x7fddc24155d0>{number = 1, name = main}
Please file a bug at https://feedbackassistant.apple.com with this warning message and any useful information you can provide.
    x testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews, (([textInputSemanticsObject waitForExistenceWithTimeout:3]) is true) failed
```

log: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8881832812710369104/+/steps/Scenario_App_Unit_Tests/0/stdout

**Comment by gaaclarke** [comment]
Here is the log from running it locally:
```
Test Case '-[TextSemanticsFocusTest testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews]' started.
    t =     0.00s Start Test at 2020-04-27 15:15:15.301
    t =     0.03s Set Up
    t =     0.05s     Open dev.flutter.Scenarios
    t =     0.08s         Launch dev.flutter.Scenarios
    t =     2.74s             Wait for accessibility to load
    t =     2.85s             Setting up automation session
    t =     5.13s             Wait for dev.flutter.Scenarios to idle
    t =     6.65s Waiting 3.0s for "flutter textfield" TextField to exist
    t =     7.66s     Checking `Expect predicate `exists == 1` for object "flutter textfield" TextField`
    t =     7.66s         Checking existence of `"flutter textfield" TextField`
    t =     7.68s Find the "flutter textfield" TextField
    t =     7.69s Waiting 3.0s for TextView to exist
    t =     8.71s     Checking `Expect predicate `exists == 1` for object TextView`
    t =     8.71s         Checking existence of `TextView`
    t =     8.72s         Capturing element debug description
    t =     9.73s     Checking `Expect predicate `exists == 1` for object TextView`
    t =     9.74s         Checking existence of `TextView`
    t =     9.75s         Capturing element debug description
    t =    10.69s Checking existence of `TextView`
    t =    10.70s Waiting 3.0s for Keyboard to exist
    t =    11.71s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    11.71s         Checking existence of `Keyboard`
    t =    11.72s         Capturing element debug description
    t =    12.73s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    12.74s         Checking existence of `Keyboard`
    t =    12.75s         Capturing element debug description
    t =    13.71s Checking existence of `Keyboard`
    t =    13.72s Tap "flutter textfield" TextField
    t =    13.72s     Wait for dev.flutter.Scenarios to idle
    t =    13.73s     Find the "flutter textfield" TextField
    t =    13.75s     Check for interrupting elements affecting "flutter textfield" TextField
    t =    13.75s     Synthesize event
    t =    13.83s     Wait for dev.flutter.Scenarios to idle
    t =    14.65s Waiting 3.0s for "focused flutter textfield" TextField to exist
    t =    15.67s     Checking `Expect predicate `exists == 1` for object "focused flutter textfield" TextField`
    t =    15.67s         Checking existence of `"focused flutter textfield" TextField`
    t =    15.70s Find the "focused flutter textfield" TextField
    t =    15.71s Waiting 3.0s for TextView to exist
    t =    16.74s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    16.74s         Checking existence of `TextView`
    t =    16.75s         Capturing element debug description
    t =    17.76s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    17.76s         Checking existence of `TextView`
    t =    17.78s         Capturing element debug description
    t =    18.72s Checking existence of `TextView`
    t =    18.74s Waiting 3.0s for Keyboard to exist
    t =    19.74s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    19.74s         Checking existence of `Keyboard`
    t =    19.76s Tear Down
Test Case '-[TextSemanticsFocusTest testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews]' passed (19.969 seconds).
```

It is a very slow test even when it works.

**Comment by gaaclarke** [comment]
I increased the timeout's to 10 seconds, there are a few timeout's that seem to run the full length of the timeout they are given:
```
Test Case '-[TextSemanticsFocusTest testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews]' started.
    t =     0.00s Start Test at 2020-04-27 15:19:08.170
    t =     0.02s Set Up
    t =     0.05s     Open dev.flutter.Scenarios
    t =     0.06s         Launch dev.flutter.Scenarios
    t =     2.92s             Setting up automation session
    t =     5.11s             Wait for dev.flutter.Scenarios to idle
    t =     6.64s Waiting 10.0s for "flutter textfield" TextField to exist
    t =     7.66s     Checking `Expect predicate `exists == 1` for object "flutter textfield" TextField`
    t =     7.66s         Checking existence of `"flutter textfield" TextField`
    t =     7.68s Find the "flutter textfield" TextField
    t =     7.69s Waiting 10.0s for TextView to exist
    t =     8.71s     Checking `Expect predicate `exists == 1` for object TextView`
    t =     8.71s         Checking existence of `TextView`
    t =     8.72s         Capturing element debug description
    t =     9.74s     Checking `Expect predicate `exists == 1` for object TextView`
    t =     9.74s         Checking existence of `TextView`
    t =     9.75s         Capturing element debug description
    t =    10.78s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    10.78s         Checking existence of `TextView`
    t =    10.79s         Capturing element debug description
    t =    11.81s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    11.81s         Checking existence of `TextView`
    t =    11.82s         Capturing element debug description
    t =    12.84s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    12.84s         Checking existence of `TextView`
    t =    12.86s         Capturing element debug description
    t =    13.88s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    13.88s         Checking existence of `TextView`
    t =    13.89s         Capturing element debug description
    t =    14.91s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    14.91s         Checking existence of `TextView`
    t =    14.92s         Capturing element debug description
    t =    15.92s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    15.93s         Checking existence of `TextView`
    t =    15.94s         Capturing element debug description
    t =    16.96s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    16.96s         Checking existence of `TextView`
    t =    16.97s         Capturing element debug description
    t =    17.70s Checking existence of `TextView`
    t =    17.71s Waiting 10.0s for Keyboard to exist
    t =    18.71s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    18.71s         Checking existence of `Keyboard`
    t =    18.72s         Capturing element debug description
    t =    19.75s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    19.75s         Checking existence of `Keyboard`
    t =    19.76s         Capturing element debug description
    t =    20.77s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    20.78s         Checking existence of `Keyboard`
    t =    20.79s         Capturing element debug description
    t =    21.81s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    21.81s         Checking existence of `Keyboard`
    t =    21.82s         Capturing element debug description
    t =    22.85s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    22.85s         Checking existence of `Keyboard`
    t =    22.86s         Capturing element debug description
    t =    23.88s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    23.88s         Checking existence of `Keyboard`
    t =    23.89s         Capturing element debug description
    t =    24.90s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    24.90s         Checking existence of `Keyboard`
    t =    24.91s         Capturing element debug description
    t =    25.93s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    25.93s         Checking existence of `Keyboard`
    t =    25.94s         Capturing element debug description
    t =    26.96s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    26.96s         Checking existence of `Keyboard`
    t =    26.97s         Capturing element debug description
    t =    27.71s Checking existence of `Keyboard`
    t =    27.72s Tap "flutter textfield" TextField
    t =    27.72s     Wait for dev.flutter.Scenarios to idle
    t =    27.74s     Find the "flutter textfield" TextField
    t =    27.75s     Check for interrupting elements affecting "flutter textfield" TextField
    t =    27.75s     Synthesize event
    t =    27.83s     Wait for dev.flutter.Scenarios to idle
    t =    28.68s Waiting 10.0s for "focused flutter textfield" TextField to exist
    t =    29.69s     Checking `Expect predicate `exists == 1` for object "focused flutter textfield" TextField`
    t =    29.69s         Checking existence of `"focused flutter textfield" TextField`
    t =    29.72s Find the "focused flutter textfield" TextField
    t =    29.74s Waiting 10.0s for TextView to exist
    t =    30.76s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    30.76s         Checking existence of `TextView`
    t =    30.78s         Capturing element debug description
    t =    31.78s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    31.78s         Checking existence of `TextView`
    t =    31.80s         Capturing element debug description
    t =    32.81s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    32.81s         Checking existence of `TextView`
    t =    32.83s         Capturing element debug description
    t =    33.85s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    33.85s         Checking existence of `TextView`
    t =    33.87s         Capturing element debug description
    t =    34.88s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    34.88s         Checking existence of `TextView`
    t =    34.90s         Capturing element debug description
    t =    35.93s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    35.93s         Checking existence of `TextView`
    t =    35.95s         Capturing element debug description
    t =    36.96s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    36.96s         Checking existence of `TextView`
    t =    36.98s         Capturing element debug description
    t =    37.99s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    37.99s         Checking existence of `TextView`
    t =    38.01s         Capturing element debug description
    t =    39.04s     Checking `Expect predicate `exists == 1` for object TextView`
    t =    39.04s         Checking existence of `TextView`
    t =    39.06s         Capturing element debug description
    t =    39.75s Checking existence of `TextView`
    t =    39.77s Waiting 10.0s for Keyboard to exist
    t =    40.78s     Checking `Expect predicate `exists == 1` for object Keyboard`
    t =    40.78s         Checking existence of `Keyboard`
    t =    40.80s Tear Down
Test Case '-[TextSemanticsFocusTest testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews]' passed (41.001 seconds).
```

**Comment by gaaclarke** [comment]
I moved the timeout to 1s and it was running fine on locally.

**Comment by gaaclarke** [comment]
cc @xster

**Comment by gaaclarke** [comment]
The timeout increase didn't solve the issue, we saw a failure again later: https://ci.chromium.org/p/flutter/builders/prod/Mac%20iOS%20Engine/4974

**Comment by chinmaygarde** [comment]
The test was disabled. Can we reword this to track re-enabling the same

**Comment by chinmaygarde** [comment]
Should we just delete the test? This has been disabled for years with no progress on enabled it.

**Comment by gaaclarke** [comment]
We should at least run it by the original author, @xster.  Here is the PR: https://github.com/flutter/engine/pull/17803

**Comment by zanderso** [comment]
From flaky issue triage: @jmagman's attempt to re-enable the test hit the same issue. Can we delete this test?

**Comment by jmagman** [comment]
It looks like the first frame of the test is supposed to fake a text field in `onBeginFrame`
https://github.com/flutter/engine/blob/2962099077b36704071802ea5595f4a016a1c214/testing/scenario_app/lib/src/send_text_focus_semantics.dart#L39-L41

Then the test taps somewhere:
https://github.com/flutter/engine/blob/2962099077b36704071802ea5595f4a016a1c214/testing/scenario_app/ios/Scenarios/ScenariosUITests/TextSemanticsFocusTest.m#L66-L67
Which triggers `onPointerDataPacket`, which "focuses" the fake text field:
https://github.com/flutter/engine/blob/2962099077b36704071802ea5595f4a016a1c214/testing/scenario_app/lib/src/send_text_focus_semantics.dart#L108-L109

When I attach, I right after the tap I can see node 0 correctly being updated with `flutter::SemanticsFlags::kIsFocused`.
https://github.com/flutter/engine/blob/2962099077b36704071802ea5595f4a016a1c214/lib/ui/semantics/semantics_update_builder.cc#L135

But then `onBeginFrame` is immediately called again (maybe because of the keyboard animating in?) and it resets node 0 back to an unfocused text field.

The test passes when I update `onBeginFrame` to only create the fake text input on the first frame, but not subsequent.

![Screen Shot 2022-02-16 at 6 52 09 PM](https://user-images.githubusercontent.com/682784/154395564-5199ba99-3823-4c5c-8383-84cd05fa977c.png)

lol

**Comment by zanderso** [comment]
The `easy fix` tag was in reference to deleting the test ;-)

**Comment by jmagman** [comment]
> The test passes when I update `onBeginFrame` to only create the fake text input on the first frame, but not subsequent.

Well, it passed on my machine at least, it still failed in CI. https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8822017057875020497/+/u/test:_Scenario_App_Integration_Tests/stdout

This seems beyond flaky, it's just failing.  @chunhtai do you have any interest in tracking this down? It tested a text field focus bug https://github.com/flutter/flutter/issues/53065.  If not, I'm going to give up and delete it, `easy fix`-style.

**Comment by chunhtai** [comment]
@jmagman I don't have the bandwidth to look into this a the moment. I am ok if we just delete it if it has been disabled for a while
